app/services/analytics.rb

```diff
@@ -121,6 +122,16 @@ def browser_attributes
     }
   end
 
+  def session_duration
+    @session[:session_started_at].present? ? Time.zone.now - session_started_at : nil
+  end
+
+  def session_started_at
+    value = @session[:session_started_at]
+    return value unless value.is_a?(String)
+    Time.zone.parse(value)
```

Comment:
I'd consider doing this as a positive check, instead of a negative check

```suggestion
    value.kind_of?(String) : Time.zone.parse(value) : value
```